### PR TITLE
Socket cleanup / user cleanup improvements, graceful shutdowns, delayed shutdowns

### PR DESCRIPTION
**Pull request by baughj**
This PR addresses the following issues and adds some new functionality

* fix: user can login multiple times under certain conditions
* fix: socket disconnects or exceptions don't always result in user removal / disconnects
* improve: shutdown control message implementation improved (needed some updating as it
  has not been tested in a very long time)
* add: logging in various places to help diagnose exceptions we keep seeing on
  Sentry (in particular in ReadCallback and OnDeath)
* add: Delayed shutdown support (e.g. n minutes) with job support
* add: GRPC call for shutting down the server, retrieving # of logged in users, and checking whether the shutdown is complete or not

With the introduction of async sending we introduced a few timing issues in various places with clients, which results in a flood of cleanup user requests which are never successful.

This refactors how we handle disconnects and logoffs to be much more aggressive (a disconnect / enqueue on a non connected client will result in an `ObjectDisposedException` populating upstream, where the client will
be forcibly removed and a user cleanup job triggered) - we also only do this (hopefully) once. This ensures we don't
waste time trying to deliver packets to a user with no client. I also refactored the cleanup job to handle edge cases where a user with no client needs to be properly cleaned up.

tl;dr: user disconnecting is now nearly instantly removed from the world; logoffs are handled the same way; socket cleanup improvements; shutdowns now cleanly logout everyone from the world and shut down the server properly; shutdowns can be queued up and will be announced every 5 minutes over 10, every 1 minute under 5, with a final notice seconds before disconnect.

### CODE REVIEW

File: hybrasyl/ControlService.cs [hunk 1]
```diff
@@ -1,92 +0,0 @@
...
- * For contributors and individual authors please refer to CONTRIBUTORS.MD.
- * 
- */
-
-using System.ServiceModel;
-
-namespace Hybrasyl
-{
-    [ServiceContract]
-    public interface IControlService
```
Comment (norrismiv) [inline hybrasyl/ControlService.cs:27]: RIP control service

File: hybrasyl/Objects/ItemObject.cs [hunk 2]
```diff
@@ -157,7 +157,7 @@ public ItemObjectType ItemObjectType
 
         public List<Xml.CastModifier> CastModifiers => Template.Properties.CastModifiers;
 
-        public uint MaximumDurability => Template.Properties.Physical.Durability;
+        public uint MaximumDurability => Template.Properties?.Physical?.Durability ?? 1;
```
Comment (norrismiv) [inline hybrasyl/Objects/ItemObject.cs:160]: We should have maximum durability = 0 if not present, as 1 shows the 1/1 on objects without it defined.

File: hybrasyl/Utility.cs [hunk 3]
```diff
@@ -192,12 +192,6 @@ static class Constants
         public const int IDLE_TIME = 60;
         public const int IDLE_CHECK = 10;
 
-        // Shutdown password
-        // This is a dirty hack until we have better role / auth support
-
-        public const string ShutdownPassword = "batterystaple8!";
```
Comment (norrismiv) [inline hybrasyl/Utility.cs:198]: RIP batterystaple8

Comment (baughj) [reply hybrasyl/Utility.cs:198]: RIP 2014-2020
press f in chat to show respects
